rest/src/main/java/discord4j/rest/request/RequestStream.java

```diff
@@ -83,16 +84,18 @@ class RequestStream {
      * 500, 502, 503 and 504). The delay is calculated using exponential backoff with jitter.
      */
     private reactor.util.retry.Retry serverErrorRetryFactory() {
-        return Retry.withThrowable(reactor.retry.Retry.onlyIf(ClientException.isRetryContextStatusCode(500, 502, 503,
-                504, 520))
-                .withBackoffScheduler(timedTaskScheduler)
-                .exponentialBackoffWithJitter(Duration.ofSeconds(2), Duration.ofSeconds(30))
-                .doOnRetry(ctx -> {
-                    if (log.isDebugEnabled()) {
-                        log.debug("Retry {} in bucket {} due to {} for {}",
-                                ctx.iteration(), id.toString(), ctx.exception().toString(), ctx.backoff());
-                    }
-                }));
+        return RetryBackoffSpec.backoff(10, Duration.ofSeconds(2))
+            .jitter(0.5)
+            .maxBackoff(Duration.ofSeconds(30))
+            .scheduler(timedTaskScheduler)
+            .doBeforeRetry(retrySignal -> {
+                if (log.isDebugEnabled()) {
+                    log.debug("Retry {} in bucket {} due to {}",
+                        retrySignal.totalRetries(),
+                        id.toString(),
+                        retrySignal.failure().toString());
+                }
+            });
```

Comment:
I had to remove the "for XXX seconds" indication here as the retrySignal doesn't provide any information about current ellapsed time. I did not find any workaround, as a first step I tried to compute the current duration but because we have a jitter it's random.

Reply:
I understand and it's fine. We could replicate it by creating a custom RetrySignal that contains the backoff value like in [GatewayRetrySpec](https://github.com/Discord4J/Discord4J/blob/master/gateway/src/main/java/discord4j/gateway/retry/GatewayRetrySpec.java) but it would be overkill for just a debug log line detail.

There's also a behavior change here to consider: 10 max attempts. Is that intentional? I agree with that too btw.